relayer/src/foreign_client.rs

```diff
@@ -244,35 +244,31 @@ impl fmt::Display for ForeignClient {
     }
 }
 
-impl ForeignClient {
+impl<Host: ChainHandle, Guest: ChainHandle> ForeignClient<Host, Guest> {
     /// Creates a new foreign client on `dst_chain`. Blocks until the client is created, or
     /// an error occurs.
     /// Post-condition: `dst_chain` hosts an IBC client for `src_chain`.
     pub fn new(
-        dst_chain: Box<dyn ChainHandle>,
-        src_chain: Box<dyn ChainHandle>,
-    ) -> Result<ForeignClient, ForeignClientError> {
+        dst_chain: Host,
+        src_chain: Guest,
```

Comment:
As mentioned in the earlier comment, I would leave these types named source (e.g., `SrcChain` or `Source` similar) & destination for the sake of keeping the code more consistent.